### PR TITLE
qemu_vm: cancel test if provided mem is less than min_mem

### PR DESCRIPTION
**Pull request by hereischen**
Signed-off-by: Haotong Chen <hachen@redhat.com>

id: 1567765

### CODE REVIEW

**Comment by hereischen** [comment]
@aliang123 please help review, thanks

File: virttest/qemu_vm.py [hunk 1]
```diff
@@ -948,9 +948,9 @@ def add_memorys(devices, params):
                 min_mem_size_m = params.get("vm_mem_minimum")
                 min_mem_size_m = float(normalize_data_size(min_mem_size_m))
                 if mem_size_m < min_mem_size_m:
-                    logging.info("Guest min memory has to be %s"
-                                 % min_mem_size_m)
-                    mem_size_m = min_mem_size_m
+                    raise exceptions.TestCancel("Guest min memory has to be %s"
```
Comment (aliang123) [inline virttest/qemu_vm.py]: try:
        self.devices, self.spice_options = self.make_create_command()
         ....      
    except Exception:
        for nic in self.virtnet:
            self._nic_tap_remove_helper(nic)
            utils_misc.log_last_traceback('Fail to create qemu command:')
            raise virt_vm.VMStartError(self.name, 'Error occurred while '
                                                       'executing make_create_command(). '
                                                       'Check the log for traceback.')

TestCancel will be handled here( in line 2725), and raise VMStartError finally; suggest not to handle TestCancel here, raise it to make case fail reason more intuitive.

**Comment by xutian** [comment]
agree with  @aliang123

File: virttest/qemu_vm.py (hunk 1 above)
Comment (hereischen) [reply virttest/qemu_vm.py]: updated, thanks.

**Comment by hereischen** [comment]
@xutian updated, thanks.

**Comment by aliang123** [comment]
# python scan_results.py 
CaseName                                 Status          StartTime       EndTime         Result     TimeElapsed
'1-smp_2.1024m.repeat1.pre_test.Host_RHEL.m7.u5.qcow2.virtio_blk.up.virtio_net.Guest.Win2016.x86_64.io-github-autotest-qemu.unattended_install.cdrom.extra_cdrom_ks.default_install.aio_threads' Finshed 2018-04-22 23:31:34 2018-04-22 23:32:04 CANCEL 29.7603
'2-smp_2.1024m.repeat1.run_test.Host_RHEL.m7.u5.qcow2.virtio_blk.up.virtio_net.Guest.Win2016.x86_64.io-github-autotest-qemu.block_hotplug_negative' Finshed 2018-04-22 23:32:37 2018-04-22 23:32:43 CANCEL 6.55467
'3-smp_2.1024m.repeat1.run_test.Host_RHEL.m7.u5.qcow2.virtio_blk.up.virtio_net.Guest.Win2016.x86_64.io-github-autotest-qemu.block_hotplug_negative.data_plane' Finshed 2018-04-22 23:32:58 2018-04-22 23:33:05 CANCEL 6.478012
Summary: 
CANCEL=3, Finshed=3

Acked-by: Aihua Liang<aliang@redhat.com>

**Pull request merged**